### DISCUSSION TITLE
[Multiplexer] Run a child process/command in a remote server

### DISCUSSION THREAD

**Discussion by abdelbk** [wezterm/wezterm#6447] [Q&A]
### What Operating System(s) are you running on?

macOS

### WezTerm version

20240203-110809-5046fc22

### Ask your question!

Hi,

I'm SSHing to a remote Linux server from a mac using `wezterm connect` to run the multiplexer.

Currently, I'm trying to implement a way to be able to open a file path from the terminal inside the current Nvim instance (after running `git status` for example) when working on the remote server.

I tried something like :
```lua
local success, stdout, stderr = wezterm.run_child_process({
  os.getenv("SHELL"),
  "-c",
  "command-to-get-current-nvim-instances",
})
```
The problem is that `wezterm.run_child_process` is executed in the domain that launched the multiplexer (my macbook). So it returns the nvim instances in macos and not in the remote server.

Is there a way to execute commands in the remote server ?

**Comment by wez** [comment]
You can use the various SpawnXXX assignment actions that accept https://wezfurlong.org/wezterm/config/lua/SpawnCommand.html to spawn programs into a domain by populating the `domain` field.  Those will spawn an associated pane.

**Reply by abdelbk** [reply]
Thank you @wez for the response. However, I would like to manipulate the result of the command inside a lua script.

**Reply by wez** [reply]
I understand, but it this is the functionality that is available.

**Reply by abdelbk** [reply]
Hi @wez. I made some progress using `SpawnCommandInNewWindow`.

Now, I'm confused about how environment variables are handled in the remote ssh domain.
I'm trying to use some environment variables (Like $HOME) in the command args but they don't seem to point the remote server.
```lua
wezterm.on("open-uri", function(window, pane, uri)
  full_path = get_file_full_path_from_uri
   window:perform_action(
    wezterm.action({
      SpawnCommandInNewWindow = {
        label = "Open file in Neovim",
        args = { "ruby", "$HOME/path/to/script/open_file_in_nvim.rb", full_path },
        domain = { DomainName = pane:get_domain_name() },
      }
    }),
    pane
  )
end)
```
The problem is that the command seems to inherit the local domain environment variables.
When I try to print the $EDITOR variable for example, it returns the one configured in my local machine even when I can see that the command is executed in the remote ssh.
I just don't understand how variables are passed.

Is there no way to execute the command in the context of the remote server ?

Thank you in advance,
